testrunner-running/src/main/scala/edu/illinois/cs/diaper/StateCapture.java

```diff
@@ -0,0 +1,1325 @@
+        }
+    }
+
+    /**
+     * Performs the diff between the two passed state maps and saves it into a file.
+     * If 'verbose' is specified, a detailed diff is also performed and appended to the file.
+     *
+     * @param  testname     name of the test currently diffing
+     * @param  beforeState  string representing the 'before' state
+     * @param  beforeRoots  set of the root static fields for the 'before' state
+     * @param  afterState   string representing the 'after' state
+     * @param  afterRoots   set of the root static fields for the 'after' state
+     * @param  fileName     name of the output file in which we save the diff
+     */
+    /* private void recordDiff(String testname, String beforeState, Set<String> beforeRoots,
+                                String afterState, Set<String> afterRoots, String fileName) {
+
+        // returns a new afterState only having the roots that are common with the beforeState
+        afterState = checkAdded(beforeState, beforeRoots, afterState, afterRoots);
+        try {
+            boolean statesAreSame = beforeState.equals(afterState);
+            // create a string builder
+            StringBuilder sb = new StringBuilder();
+            sb.append(testname);
+            sb.append(" ");
+            sb.append(statesAreSame);
+            sb.append("\n");
+
+            // if (!statesAreSame) {
+            //     writeToFile(fileName + "_" + testname.replaceAll(" ", "_")
+            //             + "_before.xml", beforeState, false);
+            //     writeToFile(fileName + "_" + testname.replaceAll(" ", "_")
+            //             + "_after.xml", afterState, false);
+            // }
+
+            // if (this.verbose) {
+            Diff diff = DiffBuilder.compare(beforeState).withTest(afterState).
+                withNodeMatcher(new DefaultNodeMatcher(
+                    // ElementSelectors.conditionalBuilder().whenElementIsNamed("entry")
+                    // .thenUse(ElementSelectors.byXPath("./key", ElementSelectors.byNameAndText))
+                    // .elseUse(ElementSelectors.byName)
+                    // .build()
+                    ElementSelectors.byName
+                ))
+                .checkForSimilar()
+                .build();
+            Iterable<Difference> differences = diff.getDifferences();
+            for (Object object : differences) {
+                Difference difference = (Difference)object;
+
+                sb.append("***********************\n");
+                sb.append(difference);
+                sb.append("\n~~~~\n");
+                makeDifferenceReport(difference, beforeState, sb);
+                sb.append("***********************\n");
+            }
+            // }
+            writeToFile(fileName, sb.toString(), true);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    } */
+
+    private void recordsubDiff(String testname, String beforeState,
```

Comment:
Since all this XML diffing does not actually require running tests (it just needs to read some file contents and compare them), we should refactor all the diffing logic out into some other helper class for others to use, so StateCapture is only used for capturing and loading state when tests are run.